python/paddle/fluid/tests/unittests/test_sparse_reshape_op.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import numpy as np
+import unittest
+
+
+class TestReshape(unittest.TestCase):
+    """
+    Test the API paddle.incubate.sparse.reshape on some sparse tensors.
+    x: sparse, out: sparse
+    """
+
+    def check_result(self, x_shape, new_shape, format):
+        """
+        x_shape: original shape
+        new_shape: new shape
+        format: "coo" or "csr"
+        Transform a sparse tensor with shape "x_shape" to
+        a sparse tensor with shape "new_shape".
+        Compare the output of paddle.reshape and the output of
+        paddle.incubate.sparse.reshape.
+        """
+        mask = paddle.randint(0, 2, x_shape).astype("float32")
+        origin_x = (paddle.rand(x_shape, dtype='float32') + 1) * mask
```

Comment:
"+ 1" here to make sure that all zero elements in "origin_x" is caused by multiplying by "mask",
or the following backward checks may fail:
```python
       np.testing.assert_allclose(sp_x.grad.to_dense().numpy(),
                                   (dense_x.grad * mask).numpy(),
                                   rtol=1e-05)
```
Here is an example:
origin_x =  [2,  0,  3, 4, 5] *  [1, 1, 0, 1, 1]
so origin_x is [2, 0, 0, 4, 5]
`sp_x.grad.to_dense().numpy()`  will  be [1, 0, 0, 1, 1] ,
while  `(dense_x.grad * mask).numpy()` will be [1, 1, 0, 1, 1], 
then the backward check fails.